core/src/main/java/com/backblaze/b2/client/B2LargeFileStorer.java

```diff
@@ -77,6 +86,28 @@
         return partStorers;
     }
 
+    private static List<Long> computeStartingBytePositions(List<B2PartStorer> partStorers) {
+        final List<Long> startingPositions = new ArrayList<>();
+
+        long cursor = 0;
+        try {
+            for (final B2PartStorer partStorer : partStorers) {
```

Comment:
one of your test cases has the same partNumber more than once.  that makes me wonder what we're assuming about part number ordering in this list...  what happens with unsorted lists? what happens with duplicate part numbers?

Reply:
I wrote some validation logic to sort and detect duplicates and gaps. I argue that a gap is a problem too because we need all of the SHA1s to finish the file.